src/utils/serverDiscoverer.js

```diff
@@ -0,0 +1,103 @@
+import { EventEmitter } from 'eventemitter3';
+import { isElectron, isCordova } from '../utils/Environment';
+
+
+class ServerDiscoverer {
+  constructor() {
+    window.addEventListener('online', this.handleNetworkChange);
+    window.addEventListener('offline', this.handleNetworkChange);
+    this.events = new EventEmitter();
+  }
+
+  on = (...args) => {
+    this.events.on(...args);
+  }
+
+  removeAllListeners = () => {
+    this.events.removeAllListeners();
+    if (isElectron() && this.browser) {
+      this.browser.removeAllListeners();
+      this.browser.stop();
+    }
+
+    if (isCordova()) {
+      this.zeroconf.unwatch('_network-canvas._tcp.', 'local.');
+    }
+  }
+
+  handleNetworkChange = () => {
+    this.checkEnvironment();
+    if (window.navigator.onLine) {
+      this.events.emit('SERVER_RESET');
```

Comment:
I assume this means that we are now relying on mdns errors and "Automatic server discovery unavailable" for "no network" warnings?

Reply:
Yeah, I changed it so that the ServerList doesn't have to be concerned with network status (ServerDiscoverer handles it, and emits events accordingly), which I think is a better design.

Reply:
I think that's a good shout!